src/borg/testsuite/helpers.py

```diff
@@ -1022,3 +1025,23 @@ def test_dash_open():
     assert dash_open('-', 'w') is sys.stdout
     assert dash_open('-', 'rb') is sys.stdin.buffer
     assert dash_open('-', 'wb') is sys.stdout.buffer
+
+
+def test_iter_separated():
+    # newline and utf-8
+    fd = StringIO('foo\nbar/baz\n나윤a선나윤선나윤선나윤선나윤선\n')
+    assert list(iter_separated(fd)) == ['foo', 'bar/baz', '나윤a선나윤선나윤선나윤선나윤선']
+
+    # null
+    fd = StringIO('foo/bar\0baz\0spam')
+    assert list(iter_separated(fd, sep='\0')) == ['foo/bar', 'baz', 'spam']
+
+    # multichar
+    fd = StringIO('foo/barSEPbazSEPspam')
+    assert list(iter_separated(fd, sep='SEP')) == ['foo/bar', 'baz', 'spam']
```

Comment:
ByteIO?

Reply:
If we agree strings should be kept in thread above then this is fine. But `iter_separated` should be agnostic of bytes vs str so adding the test can't hurt.